### PR TITLE
Deployment Issues

### PR DESCRIPTION
**Pull request by JamesIves**
**Description**
> Provide a description of what your changes do.

Resolves issues #580 #579 and #582 

Guessing the origin stuff got removed by mistake as it's referenced in the tests

**Testing Instructions**
> Give us step by step instructions on how to test your changes.

* Point workflow to `releases/v4-beta`, run the workflows listed in the issues!

**Additional Notes**
> Anything else that will help us test the pull request.

N/A

### CODE REVIEW

File: src/git.ts [hunk 1]
```diff
@@ -23,6 +23,22 @@ export async function init(action: ActionInterface): Promise<void | Error> {
...
 
+    await execute(`git remote rm origin`, action.workspace, action.silent)
+
+    try {
+      if (action.isTest === TestFlag.HAS_CHANGED_FILES) {
+        throw new Error()
+      }
+    } catch {
+      info('Attempted to remove origin but failed, continuing…')
+    }
```
Comment (Pike) [inline src/git.ts:34]: Is this actually doing something? Or did you want to put the execute into the try-catch block, too?

If so, I'd hand an extra flag to `isTest` to fake the error here?

File: src/git.ts [hunk 2]
```diff
@@ -48,7 +64,11 @@ export async function deploy(action: ActionInterface): Promise<Status> {
     const commitMessage = !isNullOrUndefined(action.commitMessage)
       ? (action.commitMessage as string)
       : `Deploying to ${action.branch}${
-          process.env.GITHUB_SHA ? ` from @ ${process.env.GITHUB_SHA}` : ''
+          process.env.GITHUB_SHA
+            ? ` from @ ${action.repositoryName ? action.repositoryName : ''}${
```
Comment (Pike) [inline src/git.ts:68]: I think you want the @ inbetween:
```suggestion
            ? ` from ${action.repositoryName ? action.repositoryName : ''}@${
```

File: src/git.ts (hunk 1 above)
Comment (JamesIves) [reply src/git.ts:34]: Ah yeah I just noticed that I didn't add the rm command in the try catch. I've updated the PR with an additional test flag for the test case too.

File: src/git.ts (hunk 2 above)
Comment (JamesIves) [reply src/git.ts]: I'm testing a change to this now, I wanna see if `from @ JamesIves/repo@123` is also valid as that way we keep the commit the messages the same.

**Comment by JamesIves** [comment]
Also resolves #581 - I silenced the output from these as the logging was quite spammy.

File: src/git.ts (hunk 2 above)
Comment (JamesIves) [reply src/git.ts]: Appears to have worked 🎉

Comment (JamesIves) [reply src/git.ts]: Just kidding, this needs to link the commit from the original repo. I've updated this once again so it links `${process.env.GITHUB_REPOSITORY}@${process.env.GITHUB_SHA}`

**Review by Pike — APPROVED** [review]
Looks good to me

**Comment by JamesIves** [comment]
Woot! I'm aiming to launch v4 Saturday morning.

**Pull request merged**